File: src/main/java/com/letgo/book/domain/BookTitle.java
package com.letgo.book.domain;

final public class BookTitle {
    private final String value;

    private BookTitle(String value) {
        this.value = value;
    }

    public static BookTitle create() {
        return new BookTitle("");
    }

    public static BookTitle create(String value) {
        return new BookTitle(value);
    }

    public String value() {
        return value;
    }

    @Override
    public boolean equals(Object anObject) {
        return anObject instanceof BookTitle && value().equals(((BookTitle) anObject).value());
    }

    @Override
    public int hashCode() {
        return value().hashCode();
    }
}

File: src/main/java/com/letgo/book/domain/BookTitle.kt
package com.letgo.book.domain

import com.letgo.shared.domain.DomainObject
import java.time.LocalDateTime

@DomainObject
data class BookTitle(
    private val value: String,
    private val createdAt: LocalDateTime,
) {
    constructor(value: String, createdAt: String) : this(value, LocalDateTime.parse(createdAt))
    constructor(value: String) : this(value, LocalDateTime.now())

    override fun toString(): String = value

    fun value(): String = value

    fun createdAt(): LocalDateTime = createdAt

    fun isNewerThan(title: BookTitle): Boolean = createdAt.isAfter(title.createdAt())
}
